src/System.Collections.NonGeneric/src/System/Collections/ArrayList.cs

```diff
@@ -64,6 +67,9 @@ public ArrayList()
         // initially empty, but will have room for the given number of elements
         // before any reallocations are required.
         // 
+        /// <summary>Initializes a new instance of the <see cref="T:System.Collections.ArrayList" /> class that is empty and has the specified initial capacity.</summary>
```

Comment:
No. The compiler will expand out whatever is necessary.  It's just like references in code: you only need to provide enough of the name so the compiler understands what you're referring to, but in the generated MSIL it uses the full name.

For example, if I write:
```C#
/// <summary>
/// A reference to <see cref="AsyncTaskMethodBuilder"/>.
/// </summary>
```
the compiler generates the following into the doc file:
```C#
<summary>
A reference to <see cref="T:System.Runtime.CompilerServices.AsyncTaskMethodBuilder"/>.
</summary>
```